### PR TITLE
feat: add dynamic wrapper for ldap plugin

### PR DESCRIPTION
**Pull request by savitojs**
## Description

* Adds dynamic wrapper for the [LDAP plugin](https://github.com/backstage/backstage/tree/master/plugins/catalog-backend-module-ldap)

## Which issue(s) does this PR fix

- Fixes [RHIDP-3105](https://issues.redhat.com/browse/RHIDP-3105)

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer

I have tested and verified that users are groups are being ingested correctly with LDAP server as per [LDAP Organizational Data](https://backstage.io/docs/integrations/ldap/org) along with their membership.

`app-config.local.yaml`

```
catalog:
  providers:
    ldapOrg:
      default:
        target: ldap://127.0.0.1:10389
        schedule:
          frequency: PT1H
          timeout: PT15M
        bind:
          dn: cn=admin,dc=savsingh,dc=com
          secret: xxxxx
        users:
          dn: ou=people,dc=savsingh,dc=com
          options:
            filter: (uid=*)
          map:
            description: title
        groups:
          dn: ou=people,dc=savsingh,dc=com
          options:
            filter: (objectClass=Group)
          map:
            description: cn
          set:
            metadata.namespace: 'default'
            spec.type: 'team'
```

**Logs:**

```bash
backend:start: {"level":"info","message":"loaded dynamic backend plugin 'backstage-plugin-catalog-backend-module-ldap-dynamic' from 'file:///home/savsingh/work/developer-portal/backstage-showcase/dynamic-plugins-root/backstage-plugin-ldap'","service":"backstage","timestamp":"2024-08-21 14:18:21"}
backend:start: {"level":"info","message":"Task worker starting: LdapOrgEntityProvider:default:refresh, {\"version\":2,\"cadence\":\"PT1H\",\"timeoutAfterDuration\":\"PT15M\"}","plugin":"catalog","service":"backstage","task":"LdapOrgEntityProvider:default:refresh","timestamp":"2024-08-21 14:18:23"}
backend:start: {"class":"LdapOrgEntityProvider","level":"info","message":"Reading LDAP users and groups","plugin":"catalog","service":"backstage","taskId":"LdapOrgEntityProvider:default:refresh","taskInstanceId":"de32830f-5830-41af-811a-016c3e322290","timestamp":"2024-08-21 14:18:23"}
backend:start: {"class":"LdapOrgEntityProvider","level":"info","message":"Read 7 LDAP users and 2 LDAP groups in 0.1 seconds. Committing...","plugin":"catalog","service":"backstage","taskId":"LdapOrgEntityProvider:default:refresh","taskInstanceId":"de32830f-5830-41af-811a-016c3e322290","timestamp":"2024-08-21 14:18:23"}
backend:start: {"class":"LdapOrgEntityProvider","level":"info","message":"Committed 7 LDAP users and 2 LDAP groups in 0.0 seconds.","plugin":"catalog","service":"backstage","taskId":"LdapOrgEntityProvider:default:refresh","taskInstanceId":"de32830f-5830-41af-811a-016c3e322290","timestamp":"2024-08-21 14:18:23"}
```

**Screenshots:**

_**Kind: Users:**_
Screenshot from 2024-08-21 14-19-45

_**Kind: Groups:**_
Screenshot from 2024-08-21 14-19-52

_**Membership:**_
Screenshot from 2024-08-21 14-24-12
Screenshot from 2024-08-21 14-20-17

_PS: These entities are from local test LDAP_

### CODE REVIEW

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/dist-dynamic/package.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  "license": "Apache-2.0",
+  "private": true,
+  "publishConfig": {
+    "access": "public",
+    "main": "dist/index.cjs.js",
+    "types": "dist/index.d.ts"
+  },
+  "backstage": {
+    "role": "backend-plugin-module",
+    "supported-versions": "1.27.7"
```
Comment (kim-tsao) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/dist-dynamic/package.json:15]: can you update this to 1.29.2?

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  "license": "Apache-2.0",
+  "private": true,
+  "publishConfig": {
+    "access": "public",
+    "main": "dist/index.cjs.js",
+    "types": "dist/index.d.ts"
+  },
+  "backstage": {
+    "role": "backend-plugin-module",
+    "supported-versions": "1.27.7"
```
Comment (kim-tsao) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/package.json:15]: can you update this to 1.29.2?

Comment (nickboldt) [reply dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/package.json:15]: Please use suggestions. 

```suggestion
    "supported-versions": "1.29.2"
```

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/dist-dynamic/package.json (hunk 1 above)
Comment (nickboldt) [reply dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-ldap-dynamic/dist-dynamic/package.json:15]:
```suggestion
    "supported-versions": "1.29.2"
```

**Review by nickboldt — APPROVED** [review]

**Pull request merged**